src/once_cell.rs

```diff
@@ -0,0 +1,92 @@
+use crate::Python;
+use std::cell::UnsafeCell;
+
+/// A write-once cell similar to `once_cell::OnceCell`.
+///
+/// Unlike `once_cell::sync` which uses locks to achieve thread safety, this implementation uses
```

Comment:
Just a nit, but it looks like that the current `OnceCell` implementation does not use `Mutex`. It adopts a lock-free approach the same as the `std::sync::Once`.

Reply:
Ah good catch - I'd looked at the `parking_lot` implementation of `OnceCell` - which does use a Mutex, so it depends on the `once_cell` feature being used :).

I'll reword this